mm2src/coins/rpc_command/get_current_mtp.rs

```diff
@@ -23,30 +23,26 @@ pub struct GetCurrentMtpResponse {
 #[serde(tag = "error_type", content = "error_data")]
 pub enum GetCurrentMtpError {
     NoSuchCoin(String),
+    #[display(fmt = "Requested coin: {}; is not supported for this action.", _0)]
     NotSupportedCoin(String),
-    UtxoRpcError(String),
+    RpcError(String),
 }
 
 impl HttpStatusCode for GetCurrentMtpError {
     fn status_code(&self) -> StatusCode {
         match self {
-            GetCurrentMtpError::NoSuchCoin(_)
-            | GetCurrentMtpError::UtxoRpcError(_)
-            | GetCurrentMtpError::NotSupportedCoin(_) => StatusCode::BAD_REQUEST,
+            GetCurrentMtpError::NoSuchCoin(_) => StatusCode::NOT_FOUND,
+            GetCurrentMtpError::RpcError(_) | GetCurrentMtpError::NotSupportedCoin(_) => StatusCode::BAD_REQUEST,
```

Comment:
`NoSuchCoin` should be `StatusCode::PRECONDITION_REQUIRED` since the coin activation is a precondition to using it in this case.
I guess the best fit for `RpcError` is `StatusCode::INTERNAL_SERVER_ERROR` although it's not implemented like this everywhere else.